src/placeos-frontend-loader/api/repositories.cr

```diff
@@ -28,16 +26,17 @@ module PlaceOS::FrontendLoader::Api
     end
 
     def self.commits(folder : String, branch : String, count : Int32 = 50, loader : Loader = Loader.instance)
-      with_query_directory(folder, loader) do |key, directory|
-        Git.repository_commits(
-          repository: key,
-          working_directory: directory,
-          count: count,
-          branch: branch
-        )
+      HashFile.config({"base_dir" => "#{loader.content_directory}/#{folder}/metadata"})
+      repo = HashFile["current_repo"].to_s
+      remote_type = HashFile["remote_type"].to_s
+      if remote_type
+        loader.set_actioner(remote_type)
+        loader.actioner.commits(repo, branch)[0...count]
+      else
+        raise Exception.new("remote_type could not be read from metadata")
```

Comment:
Could save an exception here by logging directly and returning nil (you're catching it immediately below)